Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetBrowserModel.cpp

```diff
@@ -260,34 +260,78 @@ namespace AzToolsFramework
 
                 if (Utils::FromMimeData(data, entries))
                 {
-                    for (auto entry : entries)
+                    if (entries.empty())
                     {
-                        using namespace AZ::IO;
-                        Path fromPath;
-                        Path toPath;
-                        bool isFolder{ true };
- 
-                        if (entry && (entry->RTTI_IsTypeOf(SourceAssetBrowserEntry::RTTI_Type())))
+                        return false;
+                    }
+
+                    if (entries.size() > 1)
+                    {
+                        for (auto assetEntry : entries)
                         {
-                            fromPath = entry->GetFullPath();
-                            PathView filename = fromPath.Filename();
-                            toPath = item->GetFullPath();
-                            toPath /= filename;
-                            isFolder = false;
+                            if (assetEntry->GetEntryType() == AssetBrowserEntry::AssetEntryType::Folder)
+                            {
+                                return false;
+                            }
                         }
-                        else
+                    }
+                    bool isFolder = entries[0]->GetEntryType() == AssetBrowserEntry::AssetEntryType::Folder;
+                    if (isFolder && AssetBrowserViewUtils::IsEngineOrProjectFolder(entries[0]->GetFullPath()))
+                    {
+                        return false;
+                    }
+                    using namespace AzFramework::AssetSystem;
+                    EntryTypeFilter* foldersFilter = new EntryTypeFilter();
```

Comment:
I'm curious, who gains ownership of this? does Qt manage it?

Reply:
Sorry I overlooked this bit of code, the folder filter is used when performing a move operation so that the user can pick which folder to move an asset to, but this is a move via drag and drop so the folder filter isn't needed.